src/txdb.cpp

```diff
@@ -599,7 +622,8 @@ bool CBlockTreeDB::LoadBlockIndexGuts(
                 pindexNew->nTime          = diskindex.nTime;
                 pindexNew->nBits          = diskindex.nBits;
                 pindexNew->nNonce         = diskindex.nNonce;
-                pindexNew->nSolution      = diskindex.nSolution;
+                // the Equihash solution will be loaded lazily from the dbindex entry
+                MetricsIncrementCounter("zcashd.debug.memory.trimmed_equihash_solutions");
```

Comment:
Hmm, isn't this going to count a "trim" without a corresponding "allocate" (the latter only occuring when a `CBlockIndex` is created from a `CBlockHeader`)? That makes `allocate - trim` incorrect as a measure of in-memory Equihash solutions. I think this should be removed:
```suggestion
```

Reply:
Oh I think you're right. I will check and remove this if you are.